### PR TITLE
Quality Check : add TaskName in the variable submission form

### PR DESCRIPTION
**Pull request by fviale**
- also fixes the problem when same variable is defined in tasks and job

### CODE REVIEW

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java [hunk 1]
```diff
@@ -960,6 +966,18 @@ private void build() {
         return ret;
     }
 
+    private String getTaskName(Node node) {
+        if (node.getParentNode() != null && node.getParentNode().getParentNode() != null) {
+            Node grandparentNode = node.getParentNode().getParentNode();
```
Comment (marcocast) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java]: grandparentNode :joy:

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java [hunk 2]
```diff
@@ -916,14 +917,15 @@ private void build() {
 
         Document dom = XMLParser.parse(jobDescriptor);
         NodeList variables = dom.getElementsByTagName("variable");
-        Map<String, JobVariable> ret = new HashMap<>();
+        Map<String, JobVariable> ret = new LinkedHashMap<>();
```
Comment (marcocast) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java]: why LinkedHashMap ?

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SubmitEditServlet.java [hunk 3]
```diff
@@ -246,4 +251,16 @@ private void upload(HttpServletRequest request, HttpServletResponse response) {
         }
     }
 
+    private String getTaskName(Node node) {
```
Comment (marcocast) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SubmitEditServlet.java]: This looks like the same method as the other class. Can you extract this method in an utility class and call it one time only?

Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/server/SubmitEditServlet.java]: I don't know well how to do it, as the context is completely different (client-side javascript generated by gwt / server-side servlets). But I will have a look

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java (hunk 1 above)
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java]: DOM API is in general very ugly

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java (hunk 2 above)
Comment (fviale) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java]: To preserve the order of variables as defined by the job.

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java (hunk 1 above)
Comment (marcocast) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java]: no but i like grandparent 👍

**Pull request merged**